tiledb/sm/tile/generic_tile_io.h

```diff
@@ -122,7 +122,7 @@ class GenericTileIO {
    * @param encryption_key The encryption key to use.
    * @return Status, Tile with the data.
    */
-  static Tile load(
+  static shared_ptr<Tile> load(
```

Comment:
How about we return `tdb_unique_ptr`? It can avoid the overhead of `shared_ptr` if we don't need it, and can be converted to a `shared_ptr` if we need it.